src/parser/SparqlParser.cpp

```diff
@@ -139,6 +139,10 @@ void SparqlParser::parseSelect(ParsedQuery* query) {
       query->_aliases.push_back(a);
       query->_selectedVariables.emplace_back(a._outVarName);
       _lexer.expect(")");
+    } else {
+      _lexer.accept();
+      throw ParseException("Error in SELECT: unexpected token: " +
+                           _lexer.current().raw);
```

Comment:
Argh, you beat me to it my fix was almost the same ;-)

Reply:
@niklas88 I'm currently fixing whatever I broke in the galloping join, and also fixing the normal join which had the same bug as the galloping case

Reply:
How did you even find that?

Reply:
Something had to be broken, and the values statement returned the correct number of rows, so the likely culprit was the join operation (as without the values scan and join becomes a single scan).